metadata-ingestion/src/datahub/ingestion/source/sql_common.py

```diff
@@ -213,50 +231,98 @@ def get_workunits(self) -> Iterable[SqlWorkUnit]:
+                        urn=f"urn:li:dataset:(urn:li:dataPlatform:{self.platform},{dataset_name},{self.config.env})",
+                        aspects=[],
+                    )
+                    if description is not None or properties:
+                        dataset_properties = DatasetPropertiesClass(
+                            description=description,
+                            customProperties=properties,
+                            # uri=dataset_name,
+                        )
+                        dataset_snapshot.aspects.append(dataset_properties)
+                    schema_metadata = get_schema_metadata(
+                        self.report, dataset_name, self.platform, columns
                     )
-                    dataset_snapshot.aspects.append(dataset_properties)
-                schema_metadata = get_schema_metadata(
-                    self.report, dataset_name, self.platform, columns
-                )
-                dataset_snapshot.aspects.append(schema_metadata)
-
-                mce = MetadataChangeEvent(proposedSnapshot=dataset_snapshot)
-                wu = SqlWorkUnit(id=dataset_name, mce=mce)
-                self.report.report_workunit(wu)
-                yield wu
+                    dataset_snapshot.aspects.append(schema_metadata)
+
+                    mce = MetadataChangeEvent(proposedSnapshot=dataset_snapshot)
+                    wu = SqlWorkUnit(id=dataset_name, mce=mce)
+                    self.report.report_workunit(wu)
+                    yield wu
+
+            if sql_config.include_views:
+                for view in inspector.get_view_names(schema):
+                    # TODO : change "standardize_schema_table_names" function name: it will be the same for tables and views
+                    schema, view = sql_config.standardize_schema_table_names(schema, view)
+                    dataset_name = sql_config.get_identifier(schema, view)
+                    self.report.report_entity_scanned(dataset_name, ent_type="view")
+
+                    if not sql_config.view_pattern.allowed(dataset_name):
+                        self.report.report_dropped(dataset_name)
+                        continue
+
+                    columns = inspector.get_columns(view, schema)
+                    try:
+                        view_info: dict = inspector.get_table_comment(view, schema)
+                    except NotImplementedError:
+                        description: Optional[str] = None
+                        properties: Dict[str, str] = {}
+                    else:
+                        description = view_info["text"]
+
+                        # The "properties" field is a non-standard addition to SQLAlchemy's interface.
+                        properties = view_info.get("properties", {})
+
+                    # TODO: capture inspector.get_pk_constraint
+                    # TODO: capture inspector.get_sorted_table_and_fkc_names
```

Comment:
Are these TODOs applicable for views?

Reply:
From SQLAlchemy documentation, not. Copy-paste error.